Client/Frontend/Browser/Browser.swift

```diff
@@ -53,6 +53,10 @@ class Browser: NSObject {
     /// The last title shown by this tab. Used by the tab tray to show titles for zombie tabs.
     var lastTitle: String?
 
+    /// The user agent override used for the last request, reload or navigation. Note that this property needs to
+    /// be managed by the web view's navigation delegate.
+    var lastCustomUserAgent: String? = nil
```

Comment:
I'm not fully satisfied with this state variable being managed outside of the `Browser` class. The latter would need to intercept the `WKNavigationDelegate` methods in order to manage the variable on its own but that appeared to be overly complicated to me.